### ISSUE TITLE
Allow "gh-pages" branch to be tested by TravisCI

### ISSUE THREAD

**Issue by JamesMGreene** [travis-ci/travis-ci#476]
As I primarily do all front-end development (mainly JavaScript), I typically only setup a "gh-pages" branch as the "master" branch has no value to me (for example: [JamesMGreene/jquery.textSelect](https://github.com/JamesMGreene/jquery.textSelect/)).  Rather, I want all of my stuff (code, tests, demos, docs, etc.) to be auto-hosted via my "gh-pages" project pages, so I only rely on the "gh-pages" as my main codebase — if I didn't, I would need to have a bunch of unnecessary pre-commit hooks in order to ensure all of the important pieces from the "master" branch got copied into their appropriate corresponding spot in the "gh-pages" branch for hosting.

However, after creating a ".travis.yml" file and verifying the service hook settings, my attempts to both "Test Hook" and trigger via push did not work.  After some discussion with @travis-ci and @svenfuchs on Twitter, and since GitHub does not seem to allow us to specify that the "master" branch should be the one hosted (like "gh-pages" is), they suggested I file this issue.

Long story short: I would like to run my project's [QUnit](http://docs.jquery.com/QUnit) tests using @travis-ci and [PhantomJS](http://phantomjs.org/) whenever I push to my "gh-pages" branch.  I imagine other front-end developers who follow this "gh-pages"-only repo pattern are in the same boat as I know they also have tests in their branches.
